lutris/util/graphics/drivers.py

```diff
@@ -2,83 +2,139 @@
     """Return details about a GPU"""
-    with open("/proc/driver/nvidia/gpus/%s/information" % gpu_id, encoding='utf-8') as info_file:
-        content = info_file.readlines()
-    infos = {}
-    for line in content:
+    with contextlib.suppress(OSError):
+        with open(
+            "/proc/driver/nvidia/gpus/%s/information" % gpu_id, encoding="utf-8"
+        ) as info_file:
+            content = info_file.readlines()
+        infos = {}
+        for line in content:
+            key, value = line.split(":", 1)
+            infos[key] = value.strip()
+        return infos
+    lspci_data = read_process_output(["lspci", "-v", "-s", gpu_id])
+    model_info = re.search(r"NVIDIA Corporation \w+ \[(.+?)\]", lspci_data)
+    if model_info:
+        model = model_info.group(1)
+    irq_info = re.search("IRQ ([0-9]+)", lspci_data)
+    if irq_info:
+        irq = irq_info.group(1)
+
+    info = {
+        "Model": f"NVIDIA {model}",
+        "IRQ": irq,
+        "Bus Location": gpu_id,
+    }
+    for line in lspci_data.splitlines():
+        if ":" not in line:
+            continue
         key, value = line.split(":", 1)
-        infos[key] = value.strip()
-    return infos
+        info[key.strip()] = value.strip()
+    return info
 
 
-def is_nvidia():
+def is_nvidia() -> bool:
     """Return true if the Nvidia drivers are currently in use"""
-    return os.path.exists("/proc/driver/nvidia")
+    with contextlib.suppress(OSError):
+        return os.path.exists("/proc/driver/nvidia")
+    with contextlib.suppress(OSError):
+        with open("/proc/modules") as f:
+            modules = f.read()
+        return bool(re.search(r"^nvidia ", modules, flags=re.MULTILINE))
+    glx_info = GlxInfo()
+    return "NVIDIA" in glx_info.opengl_vendor  # type: ignore[attr-defined]
 
 
-def get_gpus():
+def get_gpus() -> Iterable[str]:
     """Return GPUs connected to the system"""
     if not os.path.exists("/sys/class/drm"):
         logger.error("No GPU available on this system!")
-        return
+        return []
     try:
         cardlist = os.listdir("/sys/class/drm/")
     except PermissionError:
-        logger.error("Your system does not allow reading from /sys/class/drm, no GPU detected.")
+        logger.error(
+            "Your system does not allow reading from /sys/class/drm, no GPU detected."
+        )
         return
     for cardname in cardlist:
         if re.match(r"^card\d$", cardname):
             yield cardname
 
 
-def get_gpu_info(card):
+def get_gpu_info(card: str) -> Dict[str, str]:
     """Return information about a GPU"""
     infos = {"DRIVER": "", "PCI_ID": "", "PCI_SUBSYS_ID": ""}
     try:
-        with open("/sys/class/drm/%s/device/uevent" % card, encoding='utf-8') as card_uevent:
+        with open(
+            "/sys/class/drm/%s/device/uevent" % card, encoding="utf-8"
+        ) as card_uevent:
             content = card_uevent.readlines()
     except FileNotFoundError:
```

Comment:
And now it's FileNotFoundError?

Why isn't it Permission error? Or OSError to catch both?

Reply:
This and [the PermissionError catch above](https://github.com/lutris/lutris/pull/4866#discussion_r1199728570) were part of the existing code.

IMO the functions probably should just catch all `OSError` events, since it doesn't make much of a difference *why* there was an error, but I was opting to make minimal changes, especially since this is my first PR to Lutris. I'll adjust the whole file based on your code review.